File: Npc/11000047.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11000047: Anna
/// </summary>
public class _11000047 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0831180407000215$
    // - Can I help you?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0831180407000218$
                // - I used to love hearing the sound of bow craftsmen carving wood. The most talented of them all was Freeman. 
                return 30;
            case (30, 1):
                // $script:0831180407000219$
                // - He was $npcName:11000007[gender:1]$'s father. It's a shame an illness took him so early, but he would surely approve of the woman his daughter grew to be.
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Next,
            (30, 1) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11000047.py
""" 11000047: Anna """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407000215$
        # - Can I help you?
        return None # TODO

    def __30(self, pick: int) -> int:
        if self.index == 0:
            # $script:0831180407000218$
            # - I used to love hearing the sound of bow craftsmen carving wood. The most talented of them all was Freeman. 
            return 30
        elif self.index == 1:
            # $script:0831180407000219$
            # - He was $npcName:11000007[gender:1]$'s father. It's a shame an illness took him so early, but he would surely approve of the woman his daughter grew to be.
            return -1
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.NEXT
        elif (self.state, self.index) == (30, 1):
            return Option.CLOSE
        return Option.NONE
